src/main/java/victorzinho/music/score/generators/HexPitchPartGenerator.java

```diff
@@ -0,0 +1,156 @@
+import victorzinho.music.score.AbstractNamedMusicPart;
+import victorzinho.music.score.MusicPartGeneratorRequiringData;
+import victorzinho.music.score.MusicScore;
+
+import java.util.List;
+
+import static victorzinho.music.pointdata.PointDataFeatureCollection.ATTR_COURSE;
+import static victorzinho.music.pointdata.PointDataFeatureCollection.ATTR_SPEED;
+
+/**
+ * Generates score parts from a hex grid.
+ */
+public class HexPitchPartGenerator
+        extends AbstractNamedMusicPart
+        implements MusicPartGeneratorRequiringData<PitchClass> {
+
+    /**
+     * Obtains all the part generators for the hex grid.
+     *
+     * <ul>
+     *     <li>First one is the value for the current grid cell.</li>
+     *     <li>Second one is the value for the directly adjacent grid cell in the course direction. This is,
+     *     the cell from which the "entity" will exit the current cell with the current course; the one the "thing"
+     *     is facing.</li>
+     *     <li>3rd to 8th are all the neighbour cells, from most to least "relevant" (closest to the cell in the course direction).</li>
+     *     <li>All other n parts are obtained by the given classifiers provided by <code>extraAttributes</code></li>
+     * </ul>
+     *
+     * @param hexGridPitchClassProvider The provider of hex grid values.
+     * @return The list of part generators
+     */
+    public static List<HexPitchPartGenerator> newPartGenerators(
+            HexGridPitchClassProvider hexGridPitchClassProvider
+    ) {
+        return List.of(
+                new HexPitchPartGenerator("bass", hexGridPitchClassProvider, 0, 2),
+                new HexPitchPartGenerator("violin1", hexGridPitchClassProvider, 1, 5),
+                new HexPitchPartGenerator("violin2", hexGridPitchClassProvider, 2, 5),
+                new HexPitchPartGenerator("violin3", hexGridPitchClassProvider, 3, 4),
+                new HexPitchPartGenerator("violin4", hexGridPitchClassProvider, 4, 3),
+                new HexPitchPartGenerator("violin5", hexGridPitchClassProvider, 5, 5),
+                new HexPitchPartGenerator("violin6", hexGridPitchClassProvider, 6, 4),
+                new HexPitchPartGenerator("violin7", hexGridPitchClassProvider, 7, 3)
+        );
+    }
+
+    private final HexGridPitchClassProvider hexGridPitchClassProvider;
+    private final int pitchIndex;
+    private final int octave;
+
+    private HexPitchPartGenerator(
+            String name, HexGridPitchClassProvider hexGridPitchClassProvider, int pitchIndex, int octave
+    ) {
+        super(name);
+        this.hexGridPitchClassProvider = hexGridPitchClassProvider;
+        this.pitchIndex = pitchIndex;
+        this.octave = octave;
+    }
+
+    @Override
+    public void processNextFeature(SimpleFeature feature, PitchClass bassNote, MusicScore score) {
+
+        Double speed = (Double) feature.getAttribute(ATTR_SPEED);
+        Integer course = (Integer) feature.getAttribute(ATTR_COURSE);
```

Comment:
Document the required attributes for each part generator (if any).
Document radians vs degrees and fix marinetraffic vs yellow brick inconsistency.